internal/httpsync/httpsync.go

```diff
@@ -81,7 +81,8 @@ func (s *HttpDataSynchronizer) setHeaders(ctx context.Context, req *http.Request
 		req.SetBasicAuth(value.Username, value.Password)
 	case config.SecretTokenAuth:
 		req.Header.Set("Authorization", "Bearer "+value.Token)
+	default:
+		return fmt.Errorf("unsupported authentication type: %T", value)
 	}
-
-	return fmt.Errorf("unsupported authentication type: %T", value)
+	return nil
```

Comment:
Fun bug! After the credential was pulled out of the database properly, it ended up failing here unconditionally 😅